### PR TITLE
[FLINK-26663] Pod augmentation for the operator

### PR DESCRIPTION
**Pull request by morhidi**
- adding basic example and docs for pod augmentation using Helm + kustomize

### CODE REVIEW

**Comment by morhidi** [comment]
cc @wangyang0918 @gyfora @mbalassi

File: docs/content/docs/operations/helm.md [hunk 1]
```diff
@@ -109,3 +109,24 @@ spec:
...
+```yaml
+initContainers:
+  - name: busybox
+    image: busybox:latest
+    command: [ 'sh','-c','echo Happy Customization!' ]```
+```
+
+You can enable the example using the `--post-renderer` option when installing the Operator with Helm:
+
+```yaml
```
Comment (mbalassi) [inline docs/content/docs/operations/helm.md:129]: nit: shell

**Review by mbalassi — COMMENTED** [review]
Thanks, @morhidi. I only have a small suggestion, looks good otherwise.

**Comment by wangyang0918** [comment]
We need to clarify some behaviors.
* What's the priority of `--set` and `kustomize`?
* Could we use `kustomize` to decorate some existing container, e.g. adding resources for flink-kubernetes-operator or flink-webhook?

File: docs/content/docs/operations/helm.md (hunk 1 above)
Comment (morhidi) [reply docs/content/docs/operations/helm.md:143]: fixed, thanks

**Comment by morhidi** [comment]
> We need to clarify some behaviors.
> 
> * What's the priority of `--set` and `kustomize`?
> * Could we use `kustomize` to decorate some existing container, e.g. adding resources for flink-kubernetes-operator or flink-webhook? If it is true, then it will be great if we could have an example for setting the resources and combined with `jvmArgs`.

Thanks @wangyang0918 added a more sophisticated example that combines the Helm values with customize. PTAL

**Comment by wangyang0918** [comment]
Actually, what I mean is to configure the memory/cpu resources, which could cover FLINK-27001. Anyway, we could still do this in FLINK-27001 later.

**Review by wangyang0918 — APPROVED** [review]

**Comment by morhidi** [comment]
> Actually, what I mean is to configure the memory/cpu resources, which could cover [FLINK-27001](https://issues.apache.org/jira/browse/FLINK-27001). Anyway, we could still do this in [FLINK-27001](https://issues.apache.org/jira/browse/FLINK-27001) later.

ah ok, added those to the example too, cc @Aitozi

File: docs/content/docs/operations/helm.md [hunk 2]
```diff
@@ -109,3 +109,107 @@ spec:
...
+          resources:
+            requests:
+              memory: "0.5Gi"
+              cpu: "200m"
+            limits:
+              memory: "0.5Gi"
+              cpu: "500m"
+        - name: fluentbit
+          image: fluent/fluent-bit:1.8.12
+          command: [ 'sh','-c','/fluent-bit/bin/fluent-bit -i tail -p path=/opt/flink/log/*.log -p multiline.parser=java -o stdout' ]
```
Comment (wangyang0918) [inline docs/content/docs/operations/helm.md:193]: It seems that the `sh` is not available in the image `fluent/fluent-bit:1.8.12`.

**Review by wangyang0918 — CHANGES_REQUESTED** [review]

File: docs/content/docs/operations/helm.md (hunk 2 above)
Comment (morhidi) [reply docs/content/docs/operations/helm.md:193]: thanks @wangyang0918 bit weird, it worked on my arm64 arch laptop, changed it to the debug version in the example, anyway, to be able to attach a terminal to it. PTAL

Comment (wangyang0918) [reply docs/content/docs/operations/helm.md:193]: It is strange. I will have a try whether debug version could work.

Comment (wangyang0918) [reply docs/content/docs/operations/helm.md:193]: The flink-kubernetes-operator could run normally now. But the `fluent` does not tail the logs correctly. Do I miss some thing?

```
wangyang-pc:flink-kubernetes-operator danrtsey.wy$ kubectl logs flink-kubernetes-operator-7c6b75b9bf-nrg9x -c fluentbit
Fluent Bit v1.8.12
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/04/07 07:54:58] [ info] [engine] started (pid=8)
[2022/04/07 07:54:58] [ info] [storage] version=1.1.5, initializing...
[2022/04/07 07:54:58] [ info] [storage] in-memory
[2022/04/07 07:54:58] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/04/07 07:54:58] [ info] [cmetrics] version=0.2.2
[2022/04/07 07:54:58] [ info] [input:tail:tail.0] multiline core started
[2022/04/07 07:54:58] [ info] [sp] stream processor started
```

Comment (morhidi) [reply docs/content/docs/operations/helm.md:193]: it takes a few seconds, let me adjust the flush interval

Comment (wangyang0918) [reply docs/content/docs/operations/helm.md:193]: Could you share me some light why the `kustomize/values.yaml` does not take effect in my environment? Both the logging and the jvm args does not work for me.

```
helm install flink-kubernetes-operator helm/flink-kubernetes-operator --post-renderer examples/kustomize/render
```

Comment (morhidi) [reply docs/content/docs/operations/helm.md:193]: you have to reference it with -f values.yaml (it's just still the standard values override) it has noting to do with customise, I just put it to the example folder

```helm install flink-kubernetes-operator helm/flink-kubernetes-operator -f examples/kustomize/values.yaml --post-renderer examples/kustomize/render```

Comment (wangyang0918) [reply docs/content/docs/operations/helm.md:193]: Aha. So `examples/kustomize/values.yaml` will not be merged with `helm/values.yaml`, but just replace. Right?

Comment (morhidi) [reply docs/content/docs/operations/helm.md:193]: correct (actually it's an override in Helm too)

Comment (morhidi) [reply docs/content/docs/operations/helm.md:193]: bumped the fluent bit version to address the issue appears on latest macs 

```
no matching manifest for linux/arm64/v8 in the manifest list entries.
```

**Review by wangyang0918 — APPROVED** [review]
Verified in my minikube. This PR works really well.
+1 for merging.

**Pull request merged**